internal/controllers/app_controller.go

```diff
@@ -200,6 +201,22 @@ func (r *AppReconciler) reconcile(ctx context.Context, app *ketchv1.App) reconci
 
 	// check for canary deployment
 	if app.Spec.Canary.Active {
+		// retry until all pods for canary deployment comes to running state.
+		retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			if err := canaryPodStatus(r, app); err != nil {
+				return chart.ErrEmptyProcfile
```

Comment:
ErrEmptyProcfile?

Reply:
ah , by mistake I put it there. Fixing it. Sorry.